packages/react/src/themes/teams/components/Flex/flexStyles.ts

```diff
@@ -31,15 +31,13 @@ const flexStyles: ComponentSlotStylesInput<FlexProps, FlexVariables> = {
     }),
 
     ...(p.padding && { padding: v[p.padding] }),
-  }),
-
-  gap: ({ props: p, variables: v }) => {
-    const gapValue = v[p.gap]
 
-    return {
-      ...(p.column ? { height: gapValue } : { width: gapValue }),
-    }
-  },
+    ...(p.gap && {
+      '> *:not(:last-child)': {
```

Comment:
I tried several options for adding these styles. 
- adding `marginLeft` and `marginTop` on all elements expect the first one (this however was conflicting with the push prop on the `Flex.Item`, as it should override the same styles, but is applied first..)
- I tried applying the margins which are not used (the `marginBottom` and `marginRight`), but with this I had to find figure out which element in the Flex is the last one which can be kind a tricky (we can have null values, or even the children in the `Flex.Item` can be values - it seemed kind a risky...)
- adding `gap` property on the `Flex.Item`, which will just be propagated, and the inside the `flexItemStyles` I was adding these margins before the styles of the push prop were provided). For the other child components inside the `Flex`, I was providing the same styles. This introduced lots of duplication around the gap styles, and I didn't want to add new prop to the Flex.Item just because of the internal implementation.

That's how, I ended up with this, which seems to be the most simple solution, but I had to use this specific selector... Let me know if you have some other ideas.

Reply:
let me, please, just point out couple of problems that we would need still address - but should mention explicitly that these were the problems of the original approach with `Flex.Gap` as well. Both these problems appear when `wrap` prop is truthy for `Flex`. Lets suppose that flex direction is `row`, wrapping is happening. Then
- there is an extra gap on the right side of last element on the line - ideally we would like to omit it
![image](https://user-images.githubusercontent.com/9024564/54747915-76cdf280-4bd0-11e9-8890-92f58a6d5258.png)

- *there is no gap between flex lines - probably, eventually we might want to address this scenario